yew/src/agent/worker/public.rs

```diff
@@ -212,22 +193,18 @@ where
                 pool.remove::<RemoteAgent<AGN>>();
             }
 
+            let disconnected = ToWorker::Disconnected(self.id);
+            self.send_message(disconnected);
+
```

Comment:
nit: can you please undo this move? I prefer to keep this closure limited to pool operations